### PR TITLE
Create TypeScript Definition File

### PR DESCRIPTION
**Pull request by ExE-Boss**
This PR adds a TypeScript definition file to be able to use this in a TypeScript project (or to get intellisense in Visual Studio Code).

The base file was generated with the help of [`json-⁠schema-⁠to-⁠typescript` (on npm)](https://www.npmjs.com/package/json-schema-to-typescript), with finishing touches added manually by me.

---

I already added the `xslt` and `xpath` namespaces, as they’re getting merged in #2287 and #2026.

See also mdn/browser-compat-toolkit#11

### CODE REVIEW

**Comment by Elchi3** [comment]
If anyone knows TypeScript, I would appreciate review help. 
I guess this wouldn't do any harm for those not using TS at all, though.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+   *   (like Firefox Nightly, Chrome Canary).
+   * - `esr`: This release is an Extended Support Release.
+   * - `planned`: This release is planned in the future.
+   */
+  status: 'retired' | 'current' | 'limited' | 'beta' | 'nightly' | 'esr' | 'planned';
+}
+
+/**
+ * The `support_statement` object describes the support provided
+ * by a single browser type for the given subfeature.
```
Comment (Elchi3) [inline index.d.ts]: "sub-feature" (for consistency)

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+type VersionValue = string | boolean | null;
+
+/**
+ * The `simple_support_statement` object is the core object containing the compatibility information for a browser.
+ */
+interface SimpleSupportStatement {
+  /** The version indicating when a sub-feature has been added (and is therefore supported).
+   *
+   * The `boolean` values indicate that a sub-feature is supported
+   * (`true`, with the dditional meaning that it is unknown in which version support was added)
```
Comment (Elchi3) [inline index.d.ts]: Fix "dditional" -> "additional".

**Comment by ExE-Boss** [comment]
(In reply to @Elchi3 from https://github.com/mdn/browser-compat-data/pull/2308#pullrequestreview-147619280)
> I was trying it in VS Code and it gives me three errors and I don't know if they are relevant are not:
> - Property '__compat' of type 'CompatStatement' is not assignable to string index type 'Identifier'. (199,3)
> - Property 'browsers' of type 'Browsers' is not assignable to string index type 'Identifier'. (274, 3)
> - The expression of an export assignment must be an identifier or qualified name in an ambient context. (345, 10)

That only affects TypeScript compilation, but intellisense works fine. (It’s why I added the following comment): https://github.com/mdn/browser-compat-data/blob/0fcb9bc8ea7c4742852ee5f4f1b0cd62028d8924/index.d.ts#L343-L344

> Found two nits in the comments, but looks good to me otherwise.

I copied the descriptions from [schemas/compat-data-schema.md](https://github.com/mdn/browser-compat-data/blob/master/schemas/compat-data-schema.md) and [schemas/browsers-schema.md](https://github.com/mdn/browser-compat-data/blob/master/schemas/browsers-schema.md)..

**Pull request merged**